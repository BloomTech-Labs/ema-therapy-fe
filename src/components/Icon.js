import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Icon = (props) => {
  const { icon } = props;

  return (
    <IconStyle width="30" height="30" viewBox="0 0 500 500">
      <path d={icon} />
    </IconStyle>
  );
};

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
};

const IconStyle = styled.svg`
  border: 1px solid black;
  border-radius: 100%;
  padding: 4px;
`;

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
};

export default Icon;
