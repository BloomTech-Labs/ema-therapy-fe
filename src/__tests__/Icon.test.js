import React from 'react';
import { render } from '@testing-library/react';
import Icon from '../components/Icon';
import 'jest-styled-components';

describe('Icon component', () => {
  test('renders correctly', () => {
    expect(render(<Icon icon="testString" />)).toMatchSnapshot();
  });
});
