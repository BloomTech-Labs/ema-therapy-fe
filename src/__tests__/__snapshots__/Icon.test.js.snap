// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  border: 1px solid black;
  border-radius: 100%;
  padding: 4px;
}

<body>
    <div>
      <svg
        class="c0"
        height="30"
        viewBox="0 0 500 500"
        width="30"
      >
        <path
          d="testString"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg
      class="sc-bdVaJa hEIfhB"
      height="30"
      viewBox="0 0 500 500"
      width="30"
    >
      <path
        d="testString"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
